### PR TITLE
Multi backend

### PR DESCRIPTION
**Pull request by imwhatiam**

### CODE REVIEW

File: seahub/utils/repo.py [hunk 1]
```diff
@@ -49,3 +49,21 @@ def get_repo_shared_users(repo_id, repo_owner, include_groups=True):
...
+    storage_id = request.data.get("storage_id", None) or None
+    storage_classes = seafile_api.get_all_storage_classes()
+    storage_ids = request.user.permissions.storage_ids()
+
+    if storage_id not in \
+            [s.storage_id for s in storage_classes if s.storage_id in storage_ids ]:
+        storage_id = None
+
+    repo_id = seafile_api.create_repo(repo_name,
+            '', username, passwd, storage_id or storage_ids[0])
```
Comment (xiez) [inline seahub/utils/repo.py]: [0] == None == default ?

File: seahub/settings.py [hunk 2]
```diff
@@ -233,6 +233,9 @@
     'seahub.trusted_ip',
 )
 
+# Enabled or disable multiple storage backends.
```
Comment (llj) [inline seahub/settings.py]: ‘Enable'.

File: seahub/templates/libraries.html [hunk 3]
```diff
@@ -293,7 +300,9 @@ <h2 class="alc">{% trans "You do not have connected devices" %}</h2>
     enable_repo_snapshot_label: {% if enable_repo_snapshot_label %} true {% else %} false {% endif %},
     cur_note: {% if request.cur_note %} {'id': '{{ request.cur_note.id }}'} {% else %} null {% endif %},
     unread_notifications_request_interval: {{ unread_notifications_request_interval }},
-    enable_share_to_all_groups: {% if enable_share_to_all_groups %} true {% else %} false {% endif %}
+    enable_share_to_all_groups: {% if enable_share_to_all_groups %} true {% else %} false {% endif %},
+    enable_storage_classes: {% if enable_storage_classes %} true {% else %} false {% endif %},
+    cur_note: {% if request.cur_note %} {'id': '{{ request.cur_note.id }}'} {% else %} null {% endif %}
```
Comment (xiez) [inline seahub/templates/libraries.html]: 多余行

File: seahub/settings.py [hunk 4]
```diff
@@ -233,6 +233,9 @@
...
+ENABLE_STORAGE_CLASSES = False
+
 # Enabled or disable constance(web settings).
```
Comment (llj) [inline seahub/settings.py]: Enable.

File: seahub/templates/libraries.html (hunk 3 above)
Comment (llj) [inline seahub/templates/libraries.html]: remove it. See line 301.

File: seahub/templates/js/templates.html [hunk 5]
```diff
@@ -14,6 +14,16 @@ <h3 id="dialogTitle">{% trans "New Library"%}</h3>
         </select>
         <br/>
         <% } %>
+        <% if (app.pageOptions.is_pro && app.pageOptions.enable_storage_classes && app.pageOptions.storages.length > 1) { %>
```
Comment (llj) [inline seahub/templates/js/templates.html]: `>1` ?

File: seahub/templates/js/templates.html [hunk 6]
```diff
@@ -14,6 +14,16 @@ <h3 id="dialogTitle">{% trans "New Library"%}</h3>
...
+        <label for="storage">{% trans "Storage backend"%}</label><br/>
+        <select name="storage" id="storage" class="select">
+            <option value="<%- app.pageOptions.storages[0].storage_id %>">{% trans "Select a storage backend" %}</option>
```
Comment (llj) [inline seahub/templates/js/templates.html]: mark

File: seahub/templates/js/templates.html [hunk 7]
```diff
@@ -14,6 +14,16 @@ <h3 id="dialogTitle">{% trans "New Library"%}</h3>
...
+            <% for (var i = 0, len = app.pageOptions.storages.length; i < len; i++) { %>
+            <option value="<%- app.pageOptions.storages[i].storage_id %>"><%- app.pageOptions.storages[i].storage_name %></option>
+            <% } %>
+        </select>
+        <br/>
+        <% } %>
```
Comment (llj) [inline seahub/templates/js/templates.html]: 'storage' 放到选项的最后面吧。

**Pull request merged**